core/services/relay/evm/bindings.go

```diff
@@ -20,15 +20,19 @@ func (b Bindings) addEvent(contractName, typeName string, evt common.Hash) error
 	return nil
 }
 
-func (b Bindings) getBinding(contractName, typeName string) (*addrEvtBinding, error) {
+func (b Bindings) getBinding(contractName, typeName string, isConfig bool) (*addrEvtBinding, error) {
+	errType := types.ErrInvalidType
+	if isConfig {
+		errType = types.ErrInvalidConfig
+	}
 	typeNames, ok := b[contractName]
 	if !ok {
-		return nil, fmt.Errorf("%w: contract %s not found", types.ErrInvalidConfig, contractName)
+		return nil, fmt.Errorf("%w: contract %s not found", errType, contractName)
 	}
 
 	ae, ok := typeNames[typeName]
 	if !ok {
-		return nil, fmt.Errorf("%w: method %s not found in contract %s", types.ErrInvalidConfig, typeName, contractName)
+		return nil, fmt.Errorf("%w: method %s not found in contract %s", errType, typeName, contractName)
```

Comment:
I'm a little confused about the naming of things here.  In the error message `typeName` is referred to as a method.  But if we know it's always going to be the name of a method then how is that a type name.  Is the problem here that this `getBinding` method will be used for more than just ChainReader and the error is overly specific?

Reply:
ErrInvalidType has generally been used when you can't encode because of input errors.  I'll rename typeName here to methodName to make it clearer though.